addons/sms/models/mail_thread.py

```diff
@@ -229,8 +230,8 @@ def _message_sms(self, body, subtype_id=False, partner_ids=False, number_field=F
             subtype_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_note')
 
         return self.message_post(
-            body=plaintext2html(html2plaintext(body)), partner_ids=partner_ids or [],  # TDE FIXME: temp fix otherwise crash mail_thread.py
-            message_type='sms', subtype_id=subtype_id,
+            body=render_sms_notification(body), partner_ids=partner_ids or [],  # TDE FIXME: temp fix otherwise crash mail_thread.py
```

Comment:
Wild question (redoing the whole circle once again ?), can't we override ``message_post``, and do this if message_type is sms, so that we actually have the same behavior when calling _message_sms or message_post(message_type='sms') ? Would be a bit odd to have sms code a bit in _message_sms and message_post but maybe it would align the behavior (cfr tests) ?

Maybe in master we should more the type-sepcific processing in a method called by "message_post", so that we have a single entry point for all possible usage... if you see what I think (not sure if clear, it is monday morning).

Just proposing hein.